sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessorTest.java

```diff
@@ -140,6 +140,34 @@ void azureCoreEnvShouldBindCorrect() {
 
     }
 
+    @Test
+    void azureManagedIdentityClientIdFromEnv() {
+        Properties properties = new Properties();
+        properties.put(PROPERTY_AZURE_CLIENT_ID, "client-id-from-env");
+        PropertiesPropertySource propertiesPropertySource = new PropertiesPropertySource("test-properties", properties);
+        ConfigurableEnvironment environment = getEnvironment(propertiesPropertySource);
+        AzureGlobalProperties globalProperties = Binder.get(environment)
+            .bind(AzureGlobalProperties.PREFIX, AzureGlobalProperties.class).get();
+        assertEquals("client-id-from-env", globalProperties.getCredential().getClientId());
+        assertEquals("client-id-from-env", globalProperties.getCredential().getManagedIdentityClientId());
+        assertEquals(null, globalProperties.getCredential().getUsername());
+    }
+
+    @Test
+    void azureManagedIdentityClientIdFromUserConfig() {
+        Properties properties = new Properties();
+        properties.put(PROPERTY_AZURE_CLIENT_ID, "client-id-from-env");
+        properties.put(AzureGlobalProperties.PREFIX + ".credential.managed-identity-client-id",
```

Comment:
the value better follow the same pattern, you can use the custom-managed-identity-client-id